ros2_foxglove_bridge/tests/smoke_test.cpp

```diff
@@ -759,6 +757,7 @@ int main(int argc, char** argv) {
   const auto testResult = RUN_ALL_TESTS();
   executor->cancel();
   spinnerThread.join();
+  executor.reset();
```

Comment:
why is this necessary? this is just a shared_ptr right? so even though we already called cancel(), it is problematic to have the object's lifetime last beyond the shutdown() call? Probably could use a comment here because this line seems fairly useless otherwise.

If the lifetime matters so much, can we use `make_unique` instead of `make_shared`?

Reply:
I think the component manager was still holding on to it which caused issues. I have simplified this now quite a lot and am using the node directly instead of loading it via the component manager.